File: thesis-frontend/src/types/notification.ts
export type NotificationDeliveryState =
  | "PENDING"
  | "DELIVERED"
  | "FAILED"
  | "DISMISSED"
  | string;

export type NotificationPriority =
  | "LOW"
  | "NORMAL"
  | "HIGH"
  | "URGENT"
  | string;

export interface NotificationContentDto {
  notificationID: number;
  notificationCode: string;
  notifChannel: string;
  notifCategory: string;
  notifTitle: string;
  notifBody: string;
  notifPriority: NotificationPriority;
  actionType: string | null;
  actionUrl: string | null;
  imageUrl: string | null;
  relatedEntityName: string | null;
  relatedEntityCode: string | null;
  relatedEntityID: number | null;
  triggeredByUserCode: string | null;
  isGlobal: boolean;
  createdAt: string;
  expiresAt: string | null;
}

export interface NotificationRecipientDto {
  recipientID: number;
  notificationID: number;
  targetUserID: number;
  targetUserCode: string;
  deliveryState: NotificationDeliveryState;
  isRead: boolean;
  readAt: string | null;
  seenAt: string | null;
  dismissedAt: string | null;
  createdAt: string;
  notification: NotificationContentDto;
}

export interface NotificationUnreadCountDto {
  unreadCount: number;
}

export interface NotificationPreferenceDto {
  preferenceID: number;
  notifCategory: string;
  inAppEnabled: boolean;
  emailEnabled: boolean;
  pushEnabled: boolean;
  digestMode: string;
  quietFrom: string | null;
  quietTo: string | null;
  updatedAt: string;
}

export interface UpdateNotificationPreferenceInput {
  inAppEnabled: boolean;
  emailEnabled: boolean;
  pushEnabled: boolean;
  digestMode: string;
  quietFrom: string | null;
  quietTo: string | null;
}

export interface NotificationFeedQuery {
  page?: number;
  pageSize?: number;
  isRead?: boolean;
  notifCategory?: string;
  notifPriority?: string;
  includeDismissed?: boolean;
}

export interface NotificationCreatedEvent {
  recipientID: number;
  notificationCode: string;
  notifCategory: string;
  notifTitle: string;
  actionUrl: string | null;
  notifPriority: NotificationPriority;
  isRead: boolean;
  createdAtIso: string;
  unreadCount?: number;
}

File: ThesisManagement.Api/Models/Notification.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ThesisManagement.Api.Models
{
    [Table("NOTIFICATIONS")]
    public class Notification
    {
        [Key]
        [Column("NOTIFICATION_ID")]
        public int NotificationID { get; set; }

        [Column("NOTIFICATION_CODE")]
        public string NotificationCode { get; set; } = null!;

        [Column("NOTIF_CHANNEL")]
        public string NotifChannel { get; set; } = "IN_APP";

        [Column("NOTIF_CATEGORY")]
        public string NotifCategory { get; set; } = null!;

        [Column("NOTIF_TITLE")]
        public string NotifTitle { get; set; } = null!;

        [Column("NOTIF_BODY")]
        public string NotifBody { get; set; } = null!;

        [Column("NOTIF_PRIORITY")]
        public string NotifPriority { get; set; } = "NORMAL";

        [Column("ACTION_TYPE")]
        public string? ActionType { get; set; }

        [Column("ACTION_URL")]
        public string? ActionUrl { get; set; }

        [Column("IMAGE_URL")]
        public string? ImageUrl { get; set; }

        [Column("RELATED_ENTITY_NAME")]
        public string? RelatedEntityName { get; set; }

        [Column("RELATED_ENTITY_CODE")]
        public string? RelatedEntityCode { get; set; }

        [Column("RELATED_ENTITY_ID")]
        public int? RelatedEntityID { get; set; }

        [Column("TRIGGERED_BY_USER_ID")]
        public int? TriggeredByUserID { get; set; }

        [Column("TRIGGERED_BY_USER_CODE")]
        public string? TriggeredByUserCode { get; set; }

        [Column("IS_GLOBAL")]
        public int IsGlobal { get; set; } = 0;

        [Column("SCHEDULE_AT")]
        public DateTime? ScheduleAt { get; set; }

        [Column("EXPIRES_AT")]
        public DateTime? ExpiresAt { get; set; }

        [Column("CREATED_AT")]
        public DateTime CreatedAt { get; set; }

        [Column("UPDATED_AT")]
        public DateTime? UpdatedAt { get; set; }

        public ICollection<NotificationRecipient>? Recipients { get; set; }
    }
}
